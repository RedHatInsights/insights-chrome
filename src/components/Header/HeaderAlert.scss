.chr-c-alert {
  margin-top: var(--pf-v5-global--spacer--xs);
  // to be above the static favorite button toggle
  z-index: 10;
  .pf-v6-c-alert__title {
      padding: unset;
  }
  .pf-v6-c-alert__icon {
      padding: unset;
      background-color: unset;
  }
  .pf-v6-c-button.pf-m-plain {
    color: var(--pf-v5-global--Color--dark-200);
  }
}

.chr-c-alert-preview {
  min-width: 400px;
}
