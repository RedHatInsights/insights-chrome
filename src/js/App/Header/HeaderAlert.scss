.ins-c-alert {
    margin-top: var(--pf-global--spacer--xs);

    .pf-c-alert__title {
        padding: unset;
    }

    .pf-c-alert__icon {
        padding: unset;
        background-color: unset;
    }
}
