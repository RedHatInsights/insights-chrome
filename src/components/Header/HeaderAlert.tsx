import React, { useEffect, useState } from 'react';
import { Alert, AlertActionCloseButton, AlertVariant } from '@patternfly/react-core';
import './HeaderAlert.scss';

export type HeaderAlertProps = {
  title?: string;
  variant?: AlertVariant;
  onDismiss?: () => void;
  dismissable?: boolean;
  dismissDelay?: number;
};

const HeaderAlert = ({ title, variant = AlertVariant.info, onDismiss, dismissable = false, dismissDelay = 5000 }: HeaderAlertProps) => {
  const [alertVisible, setAlertVisible] = useState(true);
  const [timer, setTimer] = useState<NodeJS.Timer | null>(null);

  useEffect(() => {
    dismissable || createTimer();
    return () => {
      timer && clearTimeout(timer);
    };
  }, []);

  const createTimer = () => {
    timer !== null && clearTimeout(timer);
    setTimer(
      setTimeout(() => {
        setAlertVisible(false);
        if (timer) {
          clearTimeout(timer);
        }
        onDismiss && onDismiss();
      }, dismissDelay)
    );
  };

  const onClose = () => {
    onDismiss && onDismiss();
    setAlertVisible(false);
  };

  return (
    <React.Fragment>
      {alertVisible && <Alert variant={variant} title={title} className="chr-c-alert" actionClose={<AlertActionCloseButton onClose={onClose} />} />}
    </React.Fragment>
  );
};

HeaderAlert.defaultProps = {
  variant: 'info',
  dismissable: false,
  dismissDelay: 5000,
};

export default HeaderAlert;
