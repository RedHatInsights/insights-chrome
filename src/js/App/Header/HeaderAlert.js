import React, { useState, useEffect } from 'react';
import { Alert } from '@patternfly/react-core/dist/js/components/Alert/Alert';
import { AlertActionCloseButton } from '@patternfly/react-core/dist/js/components/Alert/AlertActionCloseButton';
import PropTypes from 'prop-types';
import './HeaderAlert.scss';

const HeaderAlert = ({ title, variant, onDismiss, dismissable, dismissDelay }) => {
  const [alertVisible, setAlertVisible] = useState(true);
  const [timer, setTimer] = useState(null);

  useEffect(() => {
    dismissable || createTimer();
    return () => clearTimeout(timer);
  }, []);

  const createTimer = () => {
    timer !== null && clearTimeout(timer);
    setTimer(
      setTimeout(() => {
        setAlertVisible(false);
        clearTimeout(timer);
        onDismiss && onDismiss();
      }, dismissDelay)
    );
  };

  const onClose = () => {
    onDismiss && onDismiss();
    setAlertVisible(false);
  };

  return (
    <React.Fragment>
      {alertVisible && <Alert variant={variant} title={title} className="ins-c-alert" actionClose={<AlertActionCloseButton onClose={onClose} />} />}
    </React.Fragment>
  );
};

HeaderAlert.defaultProps = {
  variant: 'info',
  dismissable: false,
  dismissDelay: 5000,
};

HeaderAlert.propTypes = {
  title: PropTypes.string,
  variant: PropTypes.string,
  onDismiss: PropTypes.func,
  dismissable: PropTypes.bool,
  dismissDelay: PropTypes.number,
};

export default HeaderAlert;
