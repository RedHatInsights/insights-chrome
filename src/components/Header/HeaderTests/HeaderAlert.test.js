import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';
import HeaderAlert from '../HeaderAlert';

describe('HeaderAlert', () => {
  it('should render correctly not dismissable', () => {
    const { container } = render(<HeaderAlert title="test" dismissDelay={0} />);
    expect(container.querySelector('div')).toMatchSnapshot();
  });

  it('should render correctly dismissable', () => {
    const onDismiss = jest.fn();
    const { container } = render(<HeaderAlert title="test" dismissable={true} onDismiss={onDismiss} />);
    expect(container.querySelector('div')).toMatchSnapshot();
  });

  it('should call onClose on user click', () => {
    const onDismiss = jest.fn();
    const { container } = render(<HeaderAlert title="test" dismissable={true} onDismiss={onDismiss} />);
    const closeButton = container.querySelector('button.pf-v6-c-button.pf-m-plain');
    expect(closeButton).toBeTruthy();
    act(() => {
      fireEvent.click(closeButton);
    });
    expect(onDismiss).toHaveBeenCalled();
  });

  it('should close itself after 5000ms delay', async () => {
    const onDismiss = jest.fn();
    jest.useFakeTimers();
    const { container, rerender } = render(<HeaderAlert title="test" dismissable={true} onDismiss={onDismiss} />);
    await act(async () => {
      jest.advanceTimersByTime(5000);
    });
    rerender();
    expect(container.querySelectorAll('div.pf-c-alert')).toHaveLength(0);
  });
});
