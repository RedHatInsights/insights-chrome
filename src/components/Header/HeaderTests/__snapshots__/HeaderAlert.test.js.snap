// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAlert should render correctly dismissable 1`] = `
<div
  aria-label="Info Alert"
  class="pf-c-alert pf-m-info chr-c-alert"
  data-ouia-component-id="OUIA-Generated-Alert-info-2"
  data-ouia-component-type="PF4/Alert"
  data-ouia-safe="true"
>
  <div
    class="pf-c-alert__icon"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
      />
    </svg>
  </div>
  <h4
    class="pf-c-alert__title"
  >
    <span
      class="pf-u-screen-reader"
    >
      Info alert:
    </span>
    test
  </h4>
  <div
    class="pf-c-alert__action"
  >
    <button
      aria-disabled="false"
      aria-label="Close Info alert: alert: test"
      class="pf-c-button pf-m-plain"
      data-ouia-component-id="OUIA-Generated-Button-plain-2"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
  
</div>
`;

exports[`HeaderAlert should render correctly not dismissable 1`] = `
<div
  aria-label="Info Alert"
  class="pf-c-alert pf-m-info chr-c-alert"
  data-ouia-component-id="OUIA-Generated-Alert-info-1"
  data-ouia-component-type="PF4/Alert"
  data-ouia-safe="true"
>
  <div
    class="pf-c-alert__icon"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
      />
    </svg>
  </div>
  <h4
    class="pf-c-alert__title"
  >
    <span
      class="pf-u-screen-reader"
    >
      Info alert:
    </span>
    test
  </h4>
  <div
    class="pf-c-alert__action"
  >
    <button
      aria-disabled="false"
      aria-label="Close Info alert: alert: test"
      class="pf-c-button pf-m-plain"
      data-ouia-component-id="OUIA-Generated-Button-plain-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 352 512"
        width="1em"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </button>
  </div>
  
</div>
`;
